### PR TITLE
Fix Init per Sender

### PR DESCRIPTION
**Pull request by ajholland**
Actually call Init Per Sender for every registered scanner for each goroutine.

### CODE REVIEW

File: processing.go [hunk 1]
```diff
@@ -91,6 +91,10 @@ func Process(mon *Monitor) {
 	//Start all the workers
 	for i := 0; i < workers; i++ {
 		go func() {
+			for _, scannerName := range orderedScanners {
+				scanner := *scanners[scannerName]
```
Comment (dadrian) [inline processing.go]: I don't think you want to dereference here. What's the type of `orderedScanners`?

Comment (ajholland) [reply processing.go]: `orderedScanners` is just strings corresponding to scanner name. `scanners` is of type `map[string]*Scanner` where Scanner is an interface, and we can't operate on a pointer to an interface.

Comment (justinbastress) [reply processing.go]: For my own education, I tried this without dereferencing it, and it fails to compile with "scanner.InitPerSender undefined (type *Scanner is pointer to interface, not interface)", which seems to make sense in this context (if it was a Scanner.(*MySQLScanner) then maybe not...

File: processing.go [hunk 2]
```diff
@@ -91,6 +91,10 @@ func Process(mon *Monitor) {
...
+				scanner.InitPerSender(i)
```
Comment (justinbastress) [inline processing.go]: The intent is to call this once for each i in [0, workers), right? If so, the outer for loop may be a problem --
 you would need to increment the shared counter inside the goroutine, right?

I confirmed by running this with a log method in InitPerSender, that it does indeed get called 1000 times, but, say, ten times for senderID=1000, but not at all for senderID=998.

Comment (dadrian) [reply processing.go]: Yeah, you'll need to pass `i` as an argument to the anonymous function:

```
for int i: =0; ....
    go func(i int) {
        for .....
    }(i) 
```

Comment (dadrian) [reply processing.go]: Good catch @justinbastress!

Comment (ajholland) [reply processing.go]: Pushed a change, thanks @justinbastress!

**Comment by dadrian** [comment]
This is covered by #23

**Pull request closed without merging**